metric_map.go

```diff
@@ -191,39 +195,41 @@ func (mm *MetricMap) receiveTimer(m *Metric, tagsKey string, now Nanotime) {
 	}
 }
 
-func (mm *MetricMap) receiveSet(m *Metric, tagsKey string, now Nanotime) {
+func (mm *MetricMap) receiveSet(m *Metric, tagsKey string) {
 	v, ok := mm.Sets[m.Name]
 	if ok {
 		s, ok := v[tagsKey]
 		if ok {
 			s.Values[m.StringValue] = struct{}{}
-			s.Timestamp = now
+			if m.Timestamp > s.Timestamp {
+				s.Timestamp = m.Timestamp
+			}
 		} else {
-			s = NewSet(now, map[string]struct{}{m.StringValue: {}}, m.Hostname, m.Tags)
+			s = NewSet(m.Timestamp, map[string]struct{}{m.StringValue: {}}, m.Hostname, m.Tags)
 		}
 		v[tagsKey] = s
 	} else {
 		mm.Sets[m.Name] = map[string]Set{
-			tagsKey: NewSet(now, map[string]struct{}{m.StringValue: {}}, m.Hostname, m.Tags),
+			tagsKey: NewSet(m.Timestamp, map[string]struct{}{m.StringValue: {}}, m.Hostname, m.Tags),
 		}
 	}
 }
 
 func (mm *MetricMap) String() string {
 	buf := new(bytes.Buffer)
 	mm.Counters.Each(func(k, tags string, counter Counter) {
-		fmt.Fprintf(buf, "stats.counter.%s: %d tags=%s\n", k, counter.Value, tags)
+		_, _ = fmt.Fprintf(buf, "stats.counter.%s: %d tags=%s\n", k, counter.Value, tags)
```

Comment:
Is this to silence linter warnings?

Reply:
Yeeaah.  I updated my goland recently, and now it has a sad about it.